data/transactions/logic/crypto_test.go

```diff
@@ -264,6 +297,44 @@ func TestEd25519VerifyBare(t *testing.T) {
 	}
 }
 
+func TestFalconVerify(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	var s crypto.FalconSeed
+	fs, err := crypto.GenerateFalconSigner(s)
+	require.NoError(t, err)
+
+	msg := "62fdfc072182654f163f5f0f9a621d729566c74d0aa413bf009c9800418c19cd"
+	data, err := hex.DecodeString(msg)
+	require.NoError(t, err)
+
+	for v := uint64(11); v <= AssemblerMaxVersion; v++ {
+		t.Run(fmt.Sprintf("v=%d", v), func(t *testing.T) {
+			ops := testProg(t, fmt.Sprintf(`arg 0; arg 1; byte 0x%s; falcon_verify`, hex.EncodeToString(fs.PublicKey[:])), v)
```

Comment:
Shouldn't assembly fail when `v < 11`?

Reply:
It does, but you are right this test is wrong. testProg ensures that the program fails up until the version below the argument. So once we have v=12, line 314 would fail. It would be upset that v11 works.

I've moved the `testProg` out of the loop, and now the loop tests all the versions starting at 11 by modifying the version byte in the program. That will be future proof.